react_juce/core/TextShadowView.h

```diff
@@ -31,44 +15,26 @@ namespace reactjuce
     {
     public:
         //==============================================================================
-        TextShadowView(View* _view) : ShadowView(_view)
-        {
-            YGNodeSetContext(yogaNode, this);
-            YGNodeSetMeasureFunc(yogaNode, measureTextNode);
-        }
+        explicit TextShadowView(View* _view);
 
         //==============================================================================
         /** Set a property on the shadow view. */
-        bool setProperty (const juce::String& name, const juce::var& value) override
-        {
-            const bool layoutPropertyWasSet = ShadowView::setProperty(name, value);
-
-            // For certain text properties we want Yoga to know that we need
-            // to measure again. For example, changing the font size.
-            if (name.compare("font-size") == 0)
-                markDirty();
-
-            return layoutPropertyWasSet;
-        }
+        bool setProperty (const juce::String& name, const juce::var& value) override;
 
         /** Override the default ShadowView behavior to explicitly error. */
-        void addChild (ShadowView* childView, int index = -1) override
-        {
-            juce::ignoreUnused (index);
-
-            if (childView != nullptr)
-            {
-                throw std::logic_error("TextShadowView cannot take children.");
-            }
-        }
+        void addChild (ShadowView* childView, int index = -1) override;
 
+        //==============================================================================
         /** Sets a flag to indicate that this node needs to be measured at the next layout pass. */
-        void markDirty()
-        {
-            YGNodeMarkDirty(yogaNode);
-        }
+        void markDirty();
+
+        //==============================================================================
 
     private:
+        //==============================================================================
+        class Pimpl;
```

Comment:
Here's where things get a bit ugly. Shadowing here. Easy enough to fix with a rename which I'm tempted to do but can we do better?